galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStoreValidator.java

```diff
@@ -6,26 +6,176 @@
 package dev.galasa.auth.couchdb.internal;
 
 import java.net.URI;
+import java.util.Random;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
 
 import dev.galasa.extensions.common.couchdb.CouchdbBaseValidator;
 import dev.galasa.extensions.common.couchdb.CouchdbException;
 import dev.galasa.extensions.common.api.HttpRequestFactory;
+import dev.galasa.framework.spi.utils.GalasaGson;
 
-public class CouchdbAuthStoreValidator extends CouchdbBaseValidator {
+public class CouchdbAuthStoreValidator extends CouchdbBaseValidator{
 
     private final Log logger = LogFactory.getLog(getClass());
+    private final GalasaGson                         gson               = new GalasaGson();
+    
+    
 
     @Override
     public void checkCouchdbDatabaseIsValid(URI couchdbUri, CloseableHttpClient httpClient, HttpRequestFactory httpRequestFactory) throws CouchdbException {
+
         // Perform the base CouchDB checks
         super.checkCouchdbDatabaseIsValid(couchdbUri, httpClient, httpRequestFactory);
-
         validateDatabasePresent(couchdbUri, CouchdbAuthStore.TOKENS_DATABASE_NAME);
+        checkTokensDesignDocument(httpClient, couchdbUri, 1);
 
         logger.debug("Auth Store CouchDB at " + couchdbUri.toString() + " validated");
+
+    }
+    
+    public void checkTokensDesignDocument( CloseableHttpClient httpClient , URI couchdbUri , int attempts) throws CouchdbException {
```

Comment:
This method is way too big. At the very least it needs splitting up.  Even wondering if a DatabaseValidator should be a separate base class, and the TokenDBValidator a subclass of that ?

Reply:
eg: Split out the GET from the PUT logic, and have a small function call them both.